File: app/src/main/java/net/tnose/app/trisquel/SelectDialogFragment.java
package net.tnose.app.trisquel;

import android.app.Dialog;
import android.content.DialogInterface;
import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.v7.app.AlertDialog;

/**
 * Created by user on 2018/06/24.
 */

public class SelectDialogFragment extends AbstractDialogFragment {
    //final String[] items = {"item_0", "item_1", "item_2"};

    @Override
    public Dialog onCreateDialog(Bundle savedInstanceState) {
        String[] items = getArguments().getStringArray("items");
        return new AlertDialog.Builder(getActivity())
                //.setTitle(getArguments().getString("title",""))
                .setItems(items, new DialogInterface.OnClickListener() {
                    @Override
                    public void onClick(DialogInterface dialog, int which) {
                        Intent data = new Intent();
                        data.putExtra("id", getArguments().getInt("id"));
                        data.putExtra("which", which);
                        notifyDialogResult(DialogInterface.BUTTON_POSITIVE, data);
                    }
                })
                .setCancelable(true)
                .create();
    }

    @Override
    public void onPause() {
        super.onPause();
        dismiss();
    }

    public static class Builder extends AbstractDialogFragment.Builder {
        @NonNull
        @Override
        protected AbstractDialogFragment build() {//build()から呼ぶとcheckArgumentsで死ぬと思う
            return new SelectDialogFragment();
        }
    }
}

File: app/src/main/java/net/tnose/app/trisquel/SelectDialogFragment.kt
package net.tnose.app.trisquel

import android.app.Dialog
import android.content.DialogInterface
import android.content.Intent
import android.os.Bundle
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.itemsIndexed
import androidx.compose.material3.AlertDialog
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.ui.Modifier
import androidx.compose.ui.platform.ComposeView
import androidx.compose.ui.unit.dp

class SelectDialogFragment : AbstractDialogFragment() {

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        val items = arguments?.getStringArray("items") ?: emptyArray()
        val ids = arguments?.getIntegerArrayList("ids")
        val id = arguments?.getInt("id") ?: -1

        val dialog = Dialog(requireContext())
        dialog.setContentView(ComposeView(requireContext()).apply {
            setContent {
                MaterialTheme {
                    AlertDialog(
                        onDismissRequest = {
                            notifyDialogCancelled()
                            dismiss()
                        },
                        title = arguments?.getString("title")?.let {
                            if (it.isNotEmpty()) {
                                { Text(text = it) }
                            } else null
                        },
                        text = {
                            LazyColumn {
                                itemsIndexed(items) { index, item ->
                                    Text(
                                        text = item,
                                        modifier = Modifier
                                            .fillMaxWidth()
                                            .clickable {
                                                val data = Intent().apply {
                                                    putExtra("id", id)
                                                    putExtra("which", index)
                                                    putExtra("which_id", ids?.get(index) ?: -1)
                                                    putExtra("which_str", item)
                                                }
                                                notifyDialogResult(DialogInterface.BUTTON_POSITIVE, data)
                                                dismiss()
                                            }
                                            .padding(16.dp),
                                        style = MaterialTheme.typography.bodyLarge
                                    )
                                }
                            }
                        },
                        confirmButton = {},
                        dismissButton = {}
                    )
                }
            }
        })
        dialog.window?.setBackgroundDrawableResource(android.R.color.transparent)
        isCancelable = true
        return dialog
    }

    override fun onPause() {
        super.onPause()
        dismiss()
    }

    class Builder : AbstractDialogFragment.Builder() {
        override fun build(): AbstractDialogFragment {
            return SelectDialogFragment()
        }
    }
}
